core/GlobalState.cc

```diff
@@ -47,6 +47,20 @@ SymbolRef GlobalState::synthesizeClass(NameRef nameId, u4 superclass, bool isMod
     return symRef;
 }
 
+namespace {
+// https://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
+u4 nextPowerOfTwo(u4 v) {
+    v--;
+    v |= v >> 1;
+    v |= v >> 2;
+    v |= v >> 4;
+    v |= v >> 8;
+    v |= v >> 16;
+    v++;
+    return v;
+}
```

Comment:
What do you think about exposing this through a header, and re-using it in the pickler?

Reply:
I can do that as a follow-on (this isn't new code; this duplication is present in master already. I only moved the function up in the file.)